System.Management.Automation 7.3.1 depends on a missing nuget package

### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Building a project dependent on System.Management.Automation 7.3.1 will complain about missing System.Runtime.CompilerServices 7.0.0-preview.3.22119.2 (which is not pushed to nuget.org at the time of writing this, only preview.2 is). It's perhaps transient ?

A quick workaround is to add https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet7/nuget/v2 to your NuGet.Config

### Expected behavior

```console
Should find the dependant System.Runtime.CompilerServices.Unsafe package, and build correctly
```


### Actual behavior

```console
error NU1102: Unable to find package System.Runtime.CompilerServices.Unsafe with version (>= 7.0.0-preview.3.22119.2)
error NU1102:   - Found 58 version(s) in nuget-org [ Nearest version: 7.0.0-preview.2.22152.2 ]
```


### Error details

_No response_

### Environment data

```powershell
7.3.1
```


### Visuals

_No response_

--- a/src/System.Management.Automation/System.Management.Automation.csproj
+++ b/src/System.Management.Automation/System.Management.Automation.csproj
@@ -40,7 +40,6 @@
     <PackageReference Include="System.DirectoryServices" Version="7.0.0" />
     <!--PackageReference Include="System.IO.FileSystem.AccessControl" Version="6.0.0-preview.5.21301.5" /-->
     <PackageReference Include="System.Management" Version="7.0.0" />
-    <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="7.0.0-preview.2.22152.2" />
     <PackageReference Include="System.Security.AccessControl" Version="6.0.2-mauipre.1.22102.15" />
     <PackageReference Include="System.Security.Cryptography.Pkcs" Version="7.0.0" />
     <PackageReference Include="System.Security.Permissions" Version="7.0.0" />

